spec/models/journal_creation_reminder_spec.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe JournalCreationReminder do
+  subject(:reminder) { described_class.create(starts_at: starts_at, ends_at: ends_at, message: "Don't forget to submit your journal before year end!") }
+  let(:starts_at) { nil }
+  let(:ends_at) { nil }
+
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:starts_at) }
+    it { is_expected.to validate_presence_of(:ends_at) }
+    it { is_expected.to validate_presence_of(:message) }
+
+    describe "starts_before_ends" do
+      context "when starts_at is before ends_at" do
+        let(:starts_at) { 3.days.ago }
+        let(:ends_at) { 3.days.from_now }
+
+        it { is_expected.to be_valid }
+      end
+
+      context "when starts_at is after ends_at" do
+        let(:starts_at) { 3.days.from_now }
+        let(:ends_at) { 3.days.ago }
+
+        it { is_expected.to be_invalid }
+
+        it "has an error message" do
+          expect(reminder.errors.messages).to eq({ starts_at: ["must be after Ending Date"] })
+        end
+      end
+
+      context "when starts_at is the same as ends_at" do
+        let(:current_time) { Time.current }
+        let(:starts_at) { current_time }
+        let(:ends_at) { current_time }
+
+        it { is_expected.to be_valid }
```

Comment:
I was surprised that this was valid, until I realized that the `ends_at` is being `.end_of_day`'d, which means that they aren't actually equal. Is this the intention of this spec? perhaps adding:
```
it "sets the end_date to the end of day" do
  expect(reminder.ends_at).to eq(current_time.end_of_day)
end
```
...so that it's clear/explicit...

but then I see that you do almost this exact thing on line 44...so maybe it's unnecessary.

Reply:
Good call